Source/WebCore/testing/Internals.cpp

```diff
@@ -7376,13 +7376,17 @@ void Internals::loadArtworkImage(String&& url, ArtworkImagePromise&& promise)
         return;
     }
     m_artworkImagePromise = makeUnique<ArtworkImagePromise>(WTFMove(promise));
-    m_artworkLoader = makeUnique<ArtworkImageLoader>(*contextDocument(), url, [this](Image* image) {
+    m_artworkLoader = makeUnique<ArtworkImageLoader>(*contextDocument(), url, [this, protectedThis = Ref { *this }](Image* image) {
         if (image) {
-            auto imageData = ImageData::create(image->width(), image->height(), { { PredefinedColorSpace::SRGB } });
-            if (!imageData.hasException())
-                m_artworkImagePromise->resolve(imageData.releaseReturnValue());
-            else
-                m_artworkImagePromise->reject(imageData.exception().code());
+            RefPtr nativeImage = image->nativeImage();
+            if (nativeImage) {
+                auto frame = WebCodecsVideoFrame::create(*contextDocument(), *nativeImage);
+                if (!frame.hasException())
+                    m_artworkImagePromise->resolve(frame.releaseReturnValue());
+                else
+                    m_artworkImagePromise->reject(frame.exception().code());
+            } else
+                m_artworkImagePromise->reject(Exception { ExceptionCode::InvalidAccessError, "No native image retrieved."_s });
```

Comment:
I would tend to write it as:
```
RefPtr document = contextDocument();
if (!document)
    return;
auto promise = std::exchange(m_artworkImagePromise, { });
RefPtr nativeImage = image ? image->nativeImage() : nullptr;
if (!nativeImage) {
    promise->reject(...);
    return;
}
promise->settle(WebCodecsVideoFrame::create(*document, *nativeImage));
```

I would tend to use `weakThis` instead of `protectedThis`.
And maybe we do not need `m_artworkImagePromise` at all and we could just capture it in the `ArtworkImageLoader` callback directly.